src/gurt/fault_inject.c

```diff
@@ -82,63 +79,117 @@ fa_op_rec_decref(struct d_hash_table *htab, d_list_t *rlink)
+/* global switch for fault injection. zero globally turns off fault injection, non-zero turns on
+ * fault injection
+ */
+static uint32_t            d_fault_inject_seed;
+static struct d_fi_gdata_t d_fi_gdata;
+static pthread_once_t      d_fi_gdata_init_once = PTHREAD_ONCE_INIT;
+
+/* Check for and set thread specific enabled  flag.  This is a single function that returns two
+ * values so it can access a single _thread local.
+ *
+ * 0 - no value set;
+ * 1 - disabled;
+ * >2 - enabled;
  */
-static uint32_t			d_fault_inject_seed;
-static struct d_fi_gdata_t	d_fi_gdata;
-static pthread_once_t		d_fi_gdata_init_once = PTHREAD_ONCE_INIT;
+static int
+fault_get_thread_enabled_h(int new_value)
+{
+	static __thread int thread_value;
+
+	if (new_value > 0)
+		thread_value = new_value;
+	return thread_value;
+}
+
+static bool
+fault_get_thread_enabled()
+{
+	int rc;
+
+	rc = fault_get_thread_enabled_h(0);
+	if (rc == 0)
+		return d_fi_gdata.dfg_thread_default;
+	else if (rc == 1)
+		return false;
+	return true;
+}
+
+static void
+fault_set_thread_enable(bool enabled)
+{
+	if (enabled)
+		fault_get_thread_enabled_h(2);
+	else
+		fault_get_thread_enabled_h(1);
+}
+
+void
+d_fault_inject_thread_enable(bool enabled)
+{
+	D_RWLOCK_RDLOCK(&d_fi_gdata.dfg_rwlock);
+	fault_set_thread_enable(enabled);
+	D_RWLOCK_UNLOCK(&d_fi_gdata.dfg_rwlock);
+}
+
+void
+d_fault_inject_thread_default_enable(bool enabled)
+{
+	D_RWLOCK_WRLOCK(&d_fi_gdata.dfg_rwlock);
+	d_fi_gdata.dfg_thread_default = enabled;
+	D_RWLOCK_UNLOCK(&d_fi_gdata.dfg_rwlock);
+}
 
 static inline int
 fault_attr_set(uint32_t fault_id, struct d_fault_attr_t fa_in, bool take_lock)
 {
-	struct d_fault_attr_t	 *fault_attr;
-	char			 *fa_argument = NULL;
-	bool			  should_free = true;
-	struct d_fault_attr	 *new_rec = NULL;
-	struct d_fault_attr	 *rec = NULL;
-	d_list_t		 *rlink = NULL;
-	int			  rc = DER_SUCCESS;
+	struct d_fault_attr_t *fault_attr;
+	char                  *fa_argument = NULL;
+	bool                   should_free = true;
+	struct d_fault_attr   *new_rec;
+	d_list_t              *rlink;
+	int                    rc = DER_SUCCESS;
 
 	D_ALLOC_PTR(new_rec);
 	if (new_rec == NULL)
 		D_GOTO(out, rc = -DER_NOMEM);
 
 	if (fa_in.fa_argument) {
-		D_STRNDUP(fa_argument, fa_in.fa_argument,
-			  FI_CONFIG_ARG_STR_MAX_LEN);
+		D_STRNDUP(fa_argument, fa_in.fa_argument, FI_CONFIG_ARG_STR_MAX_LEN);
 		if (fa_argument == NULL)
 			D_GOTO(out, rc = -DER_NOMEM);
 	}
 
 	if (take_lock)
 		D_RWLOCK_WRLOCK(&d_fi_gdata.dfg_rwlock);
 
-	rlink = d_hash_rec_find_insert(&d_fi_gdata.dfg_fa_table, &fault_id,
-				sizeof(fault_id), &new_rec->fa_link);
+	rlink = d_hash_rec_find_insert(&d_fi_gdata.dfg_fa_table, &fault_id, sizeof(fault_id),
+				       &new_rec->fa_link);
 	if (rlink == &new_rec->fa_link) {
 		fault_attr = &new_rec->fa_attr;
-		rc = D_SPIN_INIT(&fault_attr->fa_lock,
-				 PTHREAD_PROCESS_PRIVATE);
+		rc         = D_SPIN_INIT(&fault_attr->fa_lock, PTHREAD_PROCESS_PRIVATE);
```

Comment:
strange formatting here:)

Reply:
You're right, it's aligning assignments so it's the same as the line above.  I'll add a blank line and it'll put it back where you expect.